pkg/networkutils/network.go

```diff
@@ -393,6 +441,42 @@ func (n *linuxNetwork) SetupHostNetwork(vpcCIDR *net.IPNet, vpcCIDRs []*string,
 	return nil
 }
 
+func listCurrentSNATRules(ipt iptablesIface) ([]iptablesRule, error) {
+	var toClear []iptablesRule
+	log.Debug("Setup Host Network: loading existing iptables nat SNAT exclusion rules")
+
+	existingChains, err := ipt.ListChains("nat")
+	if err != nil {
+		return nil, errors.Wrap(err, "host network setup: failed to list iptables nat chains")
+	}
+	for _, chain := range existingChains {
+		if !strings.HasPrefix(chain, "AWS-SNAT-CHAIN") {
+			continue
+		}
+		rules, err := ipt.List("nat", chain)
+		if err != nil {
+			return nil, errors.Wrap(err, fmt.Sprintf("host network setup: failed to list iptables nat chain %s", chain))
+		}
+		for i, rule := range rules {
+			r := csv.NewReader(strings.NewReader(rule))
+			r.Comma = ' '
+			ruleSpec, err := r.Read()
+			if err != nil {
+				return nil, errors.Wrap(err, fmt.Sprintf("host network setup: failed to parse iptables nat chain %s rule %s", chain, rule))
+			}
+			log.Debugf("host network setup: found potentially stale SNAT rule for chain %s: %v", chain, ruleSpec)
+			toClear = append(toClear, iptablesRule{
+				name:        fmt.Sprintf("[%d] %s", i, chain),
+				shouldExist: false,
```

Comment:
It's not completely obvious that the `false` here will trigger the `ipt.Delete` on line 434 unless the newly created rules include it (`iptableRules` on line 366). Maybe just adding a comment would help, something like
```
				shouldExist: false, // To trigger ipt.Delete for stale rules
```